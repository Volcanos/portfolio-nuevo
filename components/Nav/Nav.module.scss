.menulist {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
}
.menuelem {
  margin-left: 1em;
}
.menuAnchor {
  display: inline-block;
  position: relative;
  &:hover {
    color: #e8e8e8;
    cursor: pointer;

    &:before,
    &:after {
      width: 50%;
      opacity: 1;
    }
  }

  &:before,
  &:after {
    content: "";
    position: absolute;
    bottom: 0px;
    width: 0px;
    height: 1px;
    margin: 2px 0 0;
    transition: all 0.2s ease-in-out;
    transition-duration: 0.65s;
    opacity: 0;
    background-color: #e8e8e8;
  }
  &:before {
    left: 50%;
  }
  &:after {
    right: 50%;
  }
}
