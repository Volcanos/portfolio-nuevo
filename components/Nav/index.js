import Anchor from '../Anchor';
import {menuStructure, mail} from '../Utils';
import stylesAnchor from '../Anchor/Anchor.module.scss';
import styles from './Nav.module.scss';

const handleClick = () => {
    let formedEmail = mail.reduce((a, v) => v === "#" ? a + "@" : a + v, "");
    window.open(`mailto:${formedEmail}`);
}

const Nav = () => {
    return (
        <nav>
            <ul className={styles.menulist}>
                {menuStructure.map((element, i) => {
                    return (
                        <li className={styles.menuelem} key={i}>
                            {element.url === 'handleClick' ?
                                <a className={stylesAnchor.anchor} onClick={handleClick}>
                                    {element.title}
                                </a>
                                :
                                <Anchor url={element.url} title={element.title}>
                                    {element.title}
                                </Anchor>
                            }                            
                        </li>
                    );
                })}
            </ul>
        </nav>
    )
}
export default Nav;