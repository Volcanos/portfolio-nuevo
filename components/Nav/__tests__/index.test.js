import React from 'react'
import { render } from 'enzyme';
import 'jest-styled-components'
import Nav from '../index';

describe('Nav', function () {
    it('should render all buttons', function () {
        const text = render(<Nav />).html();
        expect(text).toMatch(/Home/);
        expect(text).toMatch(/Portfolio/);
        expect(text).toMatch(/Experiencia/);
        expect(text).toMatch(/Contacto/);
    });
    it('render snapshot not change', function () {
        const tree = render(<Nav />);
        expect(tree).toMatchSnapshot();
    });
});