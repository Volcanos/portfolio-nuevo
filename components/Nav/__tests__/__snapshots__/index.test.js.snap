// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav render snapshot not change 1`] = `
.c2 {
  font-size: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
}

.c2:hover {
  color: #6f6f6f;
  cursor: pointer;
}

.c2:hover:before,
.c2:hover:after {
  width: 50%;
  opacity: 1;
}

.c2:before,
.c2:after {
  content: "";
  position: absolute;
  bottom: 0px;
  width: 0px;
  height: 1px;
  margin: 2px 0 0;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-transition-duration: 0.65s;
  transition-duration: 0.65s;
  opacity: 0;
  background-color: #6f6f6f;
}

.c2:before {
  left: 50%;
}

.c2:after {
  right: 50%;
}

.c0 {
  list-style: none;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin-left: 1em;
}

.c3 {
  display: inline-block;
  position: relative;
}

.c3:hover {
  color: #6f6f6f;
  cursor: pointer;
}

.c3:hover:before,
.c3:hover:after {
  width: 50%;
  opacity: 1;
}

.c3:before,
.c3:after {
  content: "";
  position: absolute;
  bottom: 0px;
  width: 0px;
  height: 1px;
  margin: 2px 0 0;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-transition-duration: 0.65s;
  transition-duration: 0.65s;
  opacity: 0;
  background-color: #6f6f6f;
}

.c3:before {
  left: 50%;
}

.c3:after {
  right: 50%;
}

<nav>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      <a
        class="c2"
      >
        Home
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2"
      >
        Portfolio
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c2"
      >
        Experiencia
      </a>
    </li>
    <li
      class="c1"
    >
      <a
        class="c3"
      >
        Contacto
      </a>
    </li>
  </ul>
</nav>
`;
